utils/buffer/examples/buffer_pool_exhaustion.rs

```diff
@@ -0,0 +1,65 @@
+// # Handling Buffer Pool Exhaustion and Heap Allocation
+//
+// This example demonstrates how a buffer pool is filled. The back slots of the buffer pool are
+// exhausted first, followed by the front of the buffer pool. Once both the back and front are
+// exhausted, data is allocated on the heap at a performance decrease.
+//
+// 1. Fills up the back slots of the buffer pool until they’re exhausted.
+// 2. Releases one slot to allow the buffer pool to switch to front mode.
+// 3. Fully fills the front slots of the buffer pool.
+// 4. Switches to alloc mode for direct heap allocation when both the buffer pool's back and front
+//    slots are at capacity.
+//
+// # Run
+//
+// ```
+// cargo run --example buffer_pool_exhaustion
+// ```
+
+use buffer_sv2::{Buffer, BufferPool};
+use std::collections::VecDeque;
+
+fn main() {
+    // 8 byte capacity
+    let mut buffer_pool = BufferPool::new(8);
+    let mut slices = VecDeque::new();
+
+    // Write data to fill back slots
+    for _ in 0..4 {
```

Comment:
Shouldn't this be `for _ in 0..8` ? Otherwise only 4 bytes are going to be written, but you defined a 8 bytes capacity. Maybe I'm missing something

Reply:
this is how the buffer pool evolves over as the example progresses:

```
--------  BACK MODE
a-------  BACK MODE (add "a" via loop)
aa------  BACK MODE (add "a" via loop)
aaa-----  BACK MODE (add "a" via loop)
aaaa----  BACK MODE (add "a" via loop)
-aaa----  BACK MODE (pop front)
-aaab---  BACK MODE (add "b")
-aaabc--  BACK MODE (add "c" via loop)
-aaabcc-  BACK MODE (add "c" via loop)
-aaabccc  BACK MODE (add "c" via loop)
caaabccc  BACK MODE (add "c" via loop, which gets added via front mode)
caaabccc  ALLOC MODE (add "d", allocated in a new space in the heap)
```

Reply:
@rrybarczyk it could be interesting to add this explanation above into the header comments

I'll add a suggestion